crypto/evp/evp_locl.h

```diff
@@ -95,3 +95,67 @@ void *evp_generic_fetch(OPENSSL_CTX *ctx, int operation_id,
                                             OSSL_PROVIDER *prov),
                         int (*up_ref_method)(void *),
                         void (*free_method)(void *));
+
+/* Helper macros to avoid duplicating code */
+#define getparams(_ciph,_ptr,_sz,_key,_datatype,_statvar,_fn,_args)     \
+    do {                                                                \
+        _statvar = -2;                                                  \
```

Comment:
what is this ```-2``` magic number?  why no define?

Reply:
It's used all over the place, and especially in ctrl function, to indicate unsupported controls.  So it's a "well known number"

Reply:
#define SOMENAME_RET_UNSUPPORTED     -2

Reply:
Sooo, do you want this to happen here only or all over the source? If the latter, then it's for a separate cleanup PR.

Reply:
Yeah I have been annoyed by this before - it just keeps getting worse over time  :)
Sounds like a seperate PR for later.

Reply:
Considering we use more than one standard (inspired by different other APIs over time, or what happens to be practical as the case may be), I would suggest considering the return codes from controls as a thing of its own.  They are fairly consistent.